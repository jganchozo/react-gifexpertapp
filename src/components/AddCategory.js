import React, { useState } from 'react'
import Proptypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);

        //console.log('handleInputChange called');
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        //console.log('handleSubmit', inputValue);

        if (inputValue.trim().length > 2) {
            setCategories( categories => [ inputValue, ...categories ] );        
            //setCategories( 1254885 );        
            setInputValue('');    
        }

    }

    return (
        
            <form onSubmit={ handleSubmit }>
                <p>{ inputValue }</p>
                <input  type="text"
                        value={inputValue}
                        onChange={ handleInputChange }
                />
            </form>
        
    )
}


AddCategory.propTypes = {
    setCategories: Proptypes.func.isRequired
}