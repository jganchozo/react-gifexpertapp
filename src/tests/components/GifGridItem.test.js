import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el componente GifGridItem', () => {

    const title = 'Hunter'; 
    const url = `https://media3.giphy.com/media/78ny3pDP74gDu/giphy-downsized-medium.gif?cid=395e9295xsekawpgzpc7nhl1lzm7dh4egqsp9pvhzat696eh&rid=giphy-downsized-medium.gif&ct=g`;
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('deberia devolver el componente <GifGridItem /> ', () => {
        
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de tener un parrafo con el texto', () => {
    
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        //console.log(img.props().src);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('deberia tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');

        //expect(div.props().className).toContain('animate__fadeIn');
        expect(div.props().className.includes('animate__fadeIn')).toBe(true)
        
    })
    
    
})
