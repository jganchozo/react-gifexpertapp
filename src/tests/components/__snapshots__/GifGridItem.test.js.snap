// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem deberia devolver el componente <GifGridItem />  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hunter"
    src="https://media3.giphy.com/media/78ny3pDP74gDu/giphy-downsized-medium.gif?cid=395e9295xsekawpgzpc7nhl1lzm7dh4egqsp9pvhzat696eh&rid=giphy-downsized-medium.gif&ct=g"
  />
  <p>
    Hunter
  </p>
</div>
`;
